### PR TITLE
add common components project

### PR DESCRIPTION
**Pull request by jalkire**
#### PR checklist

- [ ] Run through of all [test scenarios](https://github.com/Microsoft/accessibility-insights-windows/blob/master/docs/Scenarios.md) completed?
- [ ] Does this address an existing issue? If yes, Issue# - 
- [ ] Includes UI changes?
  - [ ] Run the production version of Accessibility Insights for Windows against a version with changes.
  - [ ] Attach any screenshots / GIF's that are applicable.

Note - After the PR has been created, certain checks will be kicked off. All of these checks must pass before a merge. 

#### Describe the change
- Added CommonUxComponents project with a few pieces of ux (messagedialog, fabricicon, progressring) to share with the extensions without coupling them to all of the other projects
- Did not yet replace existing uses of these components
- Use messagedialog in ado extension

### CODE REVIEW

File: src/AccessibilityInsights/AccessibilityInsights.csproj [hunk 1]
```diff
@@ -260,10 +260,6 @@
       <Project>{b2fa5150-470a-4864-a191-7bcae39df35a}</Project>
       <Name>Actions</Name>
     </ProjectReference>
-    <ProjectReference Include="..\AccessibilityInsights.Extensions.AzureDevOps\Extensions.AzureDevOps.csproj">
-      <Project>{58dbf9fe-0029-408f-a8a4-23ae500e938d}</Project>
-      <Name>Extensions.AzureDevOps</Name>
```
Comment (DaveTryon) [inline src/AccessibilityInsights/AccessibilityInsights.csproj]: >Extensions [](start = 12, length = 10)

Glad you caught this :)

File: src/AccessibilityInsights.CommonUxComponents/Controls/CustomControlOverridingAutomationPeer.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+{
+    /// <summary>
+    /// CustomControlOverridingAutomationPeer class
+    /// class to override LocalizedControlType and control/content element values for custom control(UserControl)
+    /// </summary>
+    public class CustomControlOverridingAutomationPeer: UserControlAutomationPeer
+    {
+        /// <summary>
+        /// Value for custom IsControlElement
+        /// </summary>
```
Comment (DaveTryon) [inline src/AccessibilityInsights.CommonUxComponents/Controls/CustomControlOverridingAutomationPeer.cs]: Comment doesn't match code. I personally prefer the full name over the abbreviated form, but that's probably not a change you're trying to introduce here

File: src/AccessibilityInsights.CommonUxComponents/Controls/CustomControlOverridingAutomationPeer.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        private bool IsControlElem;
+
+        /// <summary>
+        /// Value for custom IsContentElem
+        /// </summary>
+        private bool IsContentElem;
+
```
Comment (DaveTryon) [inline src/AccessibilityInsights.CommonUxComponents/Controls/CustomControlOverridingAutomationPeer.cs]: Remove trailing whitespaces from line 23?

File: src/AccessibilityInsights.CommonUxComponents/Controls/DualFabricIconControl.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+        }
+
+        public static void OnIconSizeBackChanged(DependencyObject o, DependencyPropertyChangedEventArgs e)
+        {
+            DualFabricIconControl sender = o as DualFabricIconControl;
+
+            if (sender != null)
+            {
+
+                sender.fabicnBack.FontSize = sender.IconSizeBack;
```
Comment (DaveTryon) [inline src/AccessibilityInsights.CommonUxComponents/Controls/DualFabricIconControl.xaml.cs]: Extra newlines are making this code look odd...

File: src/AccessibilityInsights.CommonUxComponents/Controls/DualFabricIconControl.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+        new public Brush Background
+        {
+            get { return (Brush)GetValue(BackgroundProperty); }
+
+            set { SetValue(BackgroundProperty, value); }
+        }
+
+        public static void OnBackgroundChanged(DependencyObject o, DependencyPropertyChangedEventArgs e)
+
+        {
```
Comment (DaveTryon) [inline src/AccessibilityInsights.CommonUxComponents/Controls/DualFabricIconControl.xaml.cs]: Extra newlines are making this code look odd...

File: src/AccessibilityInsights.CommonUxComponents/Controls/DualFabricIconControl.xaml.cs [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+            DualFabricIconControl sender = o as DualFabricIconControl;
+
+            if (sender != null)
+
+            {
```
Comment (DaveTryon) [inline src/AccessibilityInsights.CommonUxComponents/Controls/DualFabricIconControl.xaml.cs]: Extra newlines are making this code look odd...

**Comment by DaveTryon** [comment]
>             this.lblIcon.FontSize = 16;

Should this size be coming from a size dictionary?

---
Refers to: src/AccessibilityInsights.CommonUxComponents/Controls/FabricIconControl.xaml.cs:27 in eb4a794. [](commit_id = eb4a7940dfafee9b5e09074d980fee1ad4d34dec, deletion_comment = False)

**Comment by DaveTryon** [comment]
>         }        

Remove trailing whitespaces from line 159?

---
Refers to: src/AccessibilityInsights.CommonUxComponents/Controls/FabricIconControl.xaml.cs:159 in eb4a794. [](commit_id = eb4a7940dfafee9b5e09074d980fee1ad4d34dec, deletion_comment = False)

File: src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+            Grid.Row="0"
+            HorizontalAlignment="Center"
+            VerticalAlignment="Center"
+            Width="10"
+            Height="10"
+            Name="canv" >
+            <Ellipse Style="{DynamicResource elpsBlue}"
+            x:Name="ell1"
+                Height="4"
+                Width="4"
```
Comment (DaveTryon) [inline src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml]: >                           [](start = 25, length = 26)

Remove trailing whitespaces from line 24?

File: src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+                                </DoubleAnimationUsingKeyFrames>
+                            </Storyboard>
+                        </BeginStoryboard>
+                    </EventTrigger>
+                </Ellipse.Triggers>
+            </Ellipse>
+            <Ellipse Style="{DynamicResource elpsBlue}"
+                x:Name="ell4"
+                Height="4"
+                Width="4"
```
Comment (DaveTryon) [inline src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml]: >                           [](start = 25, length = 26)

Remove trailing whitespaces from line 131?

File: src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml.cs [hunk 9]
```diff
@@ -0,0 +1,268 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System.Windows.Controls;
```
Comment (DaveTryon) [inline src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml.cs]: >System [](start = 6, length = 6)

Please sort the using blocks alphabetically

File: src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml.cs [hunk 10]
```diff
@@ -0,0 +1,268 @@
...
+        private bool withSound;
+
+        /// <summary>
+        /// Check if Narrator is running
+        /// </summary>
+        public static bool IsInternalScreenReaderActive()
+        {
+            try
+            {
+                using (RegistryKey key = Registry.CurrentUser.OpenSubKey(Properties.Resources.ProgressRingControl_IsInternalScreenReaderActive_Software_Microsoft_Windows_NT_CurrentVersion_AccessibilityTemp))
```
Comment (DaveTryon) [inline src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml.cs]: >ProgressRingControl_IsInternalScreenReaderActive_Software_Microsoft_Windows_NT_CurrentVersion_AccessibilityTemp [](start = 94, length = 111)

It was probably unintended to put a constant registry path into a resource string, but that's probably out of scope to fix here

File: src/AccessibilityInsights.CommonUxComponents/Dialogs/MessageDialog.xaml.cs [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+
+        /// <summary>
+        /// Show message dialog, return whether user clicked "OK"
+        /// </summary>
+        /// <param name="message"></param>
+        public static bool? Show(string message)
+        {
+            var dlg = new MessageDialog() { Message = message };
+            dlg.Owner = Application.Current.MainWindow;
+
```
Comment (DaveTryon) [inline src/AccessibilityInsights.CommonUxComponents/Dialogs/MessageDialog.xaml.cs]: Remove trailing whitespaces from line 28?

File: src/AccessibilityInsights.Extensions.AzureDevOps/ConfigurationControl.xaml.cs [hunk 12]
```diff
@@ -160,7 +161,7 @@ private async void NextButton_Click(object sender, RoutedEventArgs e)
                 }
                 else
                 {
-                    //MessageDialog.Show("URL format is not valid. Example URL: https://accountname.visualstudio.com");
+                    MessageDialog.Show("URL format is not valid. Example URL: https://accountname.visualstudio.com");
```
Comment (DaveTryon) [inline src/AccessibilityInsights.Extensions.AzureDevOps/ConfigurationControl.xaml.cs]: >format [](start = 44, length = 6)

Shouldn't this be a resource string?

File: src/AccessibilityInsights.CommonUxComponents/Controls/CustomControlOverridingAutomationPeer.cs (hunk 2 above)
Comment (jalkire) [reply src/AccessibilityInsights.CommonUxComponents/Controls/CustomControlOverridingAutomationPeer.cs]: Will address this and other code-related comments in a future PR that removes the existing controls from SharedUx.

File: src/AccessibilityInsights.CommonUxComponents/Controls/CustomControlOverridingAutomationPeer.cs (hunk 3 above)
Comment (jalkire) [reply src/AccessibilityInsights.CommonUxComponents/Controls/CustomControlOverridingAutomationPeer.cs]: Removed

File: src/AccessibilityInsights.CommonUxComponents/Controls/DualFabricIconControl.xaml.cs (hunk 5 above)
Comment (jalkire) [reply src/AccessibilityInsights.CommonUxComponents/Controls/DualFabricIconControl.xaml.cs]: Agreed

**Comment by jalkire** [comment]
> > ```
> >         this.lblIcon.FontSize = 16;
> > ```
> 
> Should this size be coming from a size dictionary?
> 
> Refers to: src/AccessibilityInsights.CommonUxComponents/Controls/FabricIconControl.xaml.cs:27 in [eb4a794](https://github.com/Microsoft/accessibility-insights-windows/commit/eb4a7940dfafee9b5e09074d980fee1ad4d34dec). [](commit_id = [eb4a794](https://github.com/Microsoft/accessibility-insights-windows/commit/eb4a7940dfafee9b5e09074d980fee1ad4d34dec), deletion_comment = False)

@DaveTryon Most of this class's code was auto-generated

File: src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml (hunk 7 above)
Comment (jalkire) [reply src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml]: Removed

File: src/AccessibilityInsights.CommonUxComponents/Controls/DualFabricIconControl.xaml.cs (hunk 4 above)
Comment (jalkire) [reply src/AccessibilityInsights.CommonUxComponents/Controls/DualFabricIconControl.xaml.cs]: Fixed

File: src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml (hunk 8 above)
Comment (jalkire) [reply src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml]: Removed

File: src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml.cs (hunk 9 above)
Comment (jalkire) [reply src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml.cs]: Sorted

File: src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml.cs (hunk 10 above)
Comment (jalkire) [reply src/AccessibilityInsights.CommonUxComponents/Controls/ProgressRingControl.xaml.cs]: Yes and yes

File: src/AccessibilityInsights.CommonUxComponents/Dialogs/MessageDialog.xaml.cs (hunk 11 above)
Comment (jalkire) [reply src/AccessibilityInsights.CommonUxComponents/Dialogs/MessageDialog.xaml.cs]: Removed

File: src/AccessibilityInsights.Extensions.AzureDevOps/ConfigurationControl.xaml.cs (hunk 12 above)
Comment (jalkire) [reply src/AccessibilityInsights.Extensions.AzureDevOps/ConfigurationControl.xaml.cs]: Yes, will address in future PR

**Pull request merged**